tests/unit/test_project.py

```diff
@@ -164,13 +163,14 @@ def test_project_parts_validation(yaml_data):
     )
 
 
-def test_project_load(yaml_data):
-    Path("rockcraft.yaml").write_text(
+def test_project_load(yaml_data, tmp_path):
+    rockcraft_file = tmp_path / "rockcraft.yaml"
+    rockcraft_file.write_text(
         textwrap.dedent(json.dumps(yaml_data)),
         encoding="utf-8",
     )
 
-    project = load_project("rockcraft.yaml")
+    project = load_project(str(rockcraft_file))
```

Comment:
We can refactor `load_project` later to accept a Path argument.

Reply:
Yeap it's a good idea, I think the function already works with `Paths` because it just forwards the argument to `open`, but I thought I'd try to minimize the number of touched files on this first PR